jsonschema/_utils.py

```diff
@@ -38,6 +38,22 @@ def __repr__(self):
         return repr(self.store)
 
 
+class Cache(object):
```

Comment:
Sadly with `lru_cache` the benchmark is back to ~21ms (it's ~19.5ms with my custom Cache object).  I guess that is good enough though. I understand wanting to use something that already exists.